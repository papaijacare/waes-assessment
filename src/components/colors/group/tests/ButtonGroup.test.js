import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { ButtonGroup, ColorButton } from '../..'

describe('ButtonGroup', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<ButtonGroup role="mock" />)
    const component = wrapper.dive()

    expect(toJson(component)).toMatchSnapshot()
  })

  test('it has 3 buttons', () => {
    const wrapper = shallow(<ButtonGroup role="mock" />)
    const component = wrapper.dive()

    expect(component.find(ColorButton)).toHaveLength(3)
  })
})
