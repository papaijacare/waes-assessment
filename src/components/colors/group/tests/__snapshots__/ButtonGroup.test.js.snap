// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<div
  className="ui buttons"
>
  <Connect(ColorButton)
    color="red"
    key="mock-red"
    role="mock"
  />
  <Connect(ColorButton)
    color="yellow"
    key="mock-yellow"
    role="mock"
  />
  <Connect(ColorButton)
    color="green"
    key="mock-green"
    role="mock"
  />
</div>
`
